### PR TITLE
Implement ACL GENPASS

### PR DESCRIPTION
**Pull request by prvyk**
The PR implements ACL GENPASS + tests. It's very easy to set up when .NET already implements the random routine for us and there's no DB access.

### CODE REVIEW

File: libs/server/Resp/ACLCommands.cs [hunk 1]
```diff
@@ -367,6 +367,38 @@ private bool NetworkAclSave()
...
+            if (parseState.Count > 1)
+            {
+                return AbortWithWrongNumberOfArguments("acl|genpass");
+            }
+
+            // Default length
+            var length = 64;
+            if (parseState.Count == 1)
+            {
+                if (!parseState.TryGetLong(0, out var chars))
```
Comment (TalZaccai) [inline libs/server/Resp/ACLCommands.cs]: nit: rename to bits, as this is not the actual # of chars

File: libs/server/Resp/ACLCommands.cs [hunk 2]
```diff
@@ -367,6 +367,38 @@ private bool NetworkAclSave()
...
+                {
+                    return AbortWithErrorMessage(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER);
+                }
+
+                if ((chars <= 0) || (chars > 4096))
+                {
+                    return AbortWithErrorMessage("ERR ACL GENPASS argument must be the number of bits for the output password, a positive number up to 4096"u8);
+                }
+
+                length = (chars % 4 == 0) ? (int)chars / 4 : (int)chars / 4 + 1;
```
Comment (TalZaccai) [inline libs/server/Resp/ACLCommands.cs]: Add a comment: ```// Bits gets rounded to the next multiple of 4 and converted to bytes to get the password length```

Comment (TalZaccai) [inline libs/server/Resp/ACLCommands.cs]: Might be a bit clearer: ```length = (int)(chars / 4) + (chars % 4 == 0 ? 0 : 1);```

File: test/Garnet.test/Resp/ACL/BasicTests.cs [hunk 3]
```diff
@@ -111,6 +111,19 @@ public async Task BasicUsersTest()
             ClassicAssert.Contains("default", users);
         }
 
+        [Test]
+        public async Task BasicGenPassTest()
+        {
```
Comment (TalZaccai) [inline test/Garnet.test/Resp/ACL/BasicTests.cs:116]: Need to test edge cases and error conditions as well (e.g. 0, 4096, 4097, abcd, ...)

Comment (prvyk) [reply test/Garnet.test/Resp/ACL/BasicTests.cs:116]: Done. Couldn't use Assert.Throws, the delegate NUnit expects isn't compatible with GarnetClientSession.

File: libs/server/Resp/ACLCommands.cs (hunk 2 above)
Comment (prvyk) [reply libs/server/Resp/ACLCommands.cs]: I was conflicted on what's clearer and preferred doing the check outside. But inside will do as well. Done.

File: test/Garnet.test/Resp/ACL/BasicTests.cs (hunk 3 above)
Comment (TalZaccai) [reply test/Garnet.test/Resp/ACL/BasicTests.cs:116]: Not sure I understand... Why not do for example: 
```cs
Assert.ThrowsAsync<Exception>(async () => await c.ExecuteAsync("ACL", "GENPASS", "abcd"),
    Encoding.ASCII.GetString(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER));
```
?

Comment (prvyk) [reply test/Garnet.test/Resp/ACL/BasicTests.cs:116]: Ah, I forgot ThrowAsync existed... Done.

**Review by TalZaccai — APPROVED** [review]
